debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlDatabaseSchema.java

```diff
@@ -290,7 +292,10 @@ private void emitChangeEvent(MySqlOffsetContext offset, List<SchemaChangeEvent>
     }
 
     private boolean acceptableDatabase(final String databaseName) {
-        return filters.databaseFilter().test(databaseName) || databaseName == null || databaseName.isEmpty();
+        return !storeOnlyCapturedTables()
+                || filters.databaseFilter().test(databaseName)
```

Comment:
Is `test()` null-safe, or should the check for null be pulled ahead?

Reply:
It is null safe